sql/src/main/java/org/apache/druid/sql/calcite/schema/SystemSchema.java

```diff
@@ -682,6 +694,49 @@ public void close()
     );
   }
 
+  private static void checkResponse(
+      ListenableFuture<InputStream> future,
+      Request request,
+      BytesAccumulatingResponseHandler responseHandler
+  )
+  {
+    Futures.addCallback(
+        future,
+        new FutureCallback<InputStream>()
+        {
+          @Override
+          public void onSuccess(@Nullable InputStream result)
+          {
+            if (responseHandler.getStatus() == HttpServletResponse.SC_NO_CONTENT) {
```

Comment:
Should `HttpServletResponse.SC_NO_CONTENT` be a failure as well?

Reply:
`HttpServletResponse.SC_NO_CONTENT` corresponds to a `204` http status code, and 2xx status codes generally represent success, so I think it should not be a failure. `204` means "The server successfully processed the request and is not returning any content" from [wikipedia](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes)

Reply:
Did you observe that status code being returned on any API calls? It looks like getDatabaseSegments in MetadataResource will always return at least an empty JSON array with status OK, so getting SC_NO_CONTENT seems like an error.

Reply:
This part changed with the latest update, I am not checking this status code specifically anymore, so this can be closed i think.